src/git/repo.rs

```diff
@@ -1049,6 +1244,12 @@ impl Index {
             file_mode: entry.mode,
         })
     }
+
+    pub fn from_tree(tree: &Tree) -> eyre::Result<Self> {
```

Comment:
As per above comment, I don't think we need this, and it would be good to omit from our interfaces since it's likely to be slow on large repositories.